cmake/projects/ncursesw/hunter.cmake

```diff
@@ -0,0 +1,43 @@
+# Copyright (c) 2016-2019, Ruslan Baratov
+# All rights reserved.
+
+# !!! DO NOT PLACE HEADER GUARDS HERE !!!
+
+include(hunter_add_version)
+include(hunter_cacheable)
+include(hunter_cmake_args)
+include(hunter_configuration_types)
+include(hunter_download)
+include(hunter_pick_scheme)
+
+hunter_add_version(
+    PACKAGE_NAME
+    ncursesw
+    VERSION
+    6.1
+    URL
+    "https://ftp.gnu.org/gnu/ncurses/ncurses-6.1.tar.gz"
+    SHA1
+    57acf6bc24cacd651d82541929f726f4def780cc
+)
+
+hunter_configuration_types(ncursesw CONFIGURATION_TYPES Release)
+hunter_pick_scheme(DEFAULT url_sha1_autotools)
+hunter_cmake_args(
+    ncursesw
+    CMAKE_ARGS
+        EXTRA_FLAGS=--enable-ext-mouse --enable-sigwinch --enable-ext-colors --enable-widec --enable-pc-files "--with-pkg-config-libdir=@HUNTER_PACKAGE_INSTALL_PREFIX@/lib/pkgconfig"
```

Comment:
> "--with-pkg-config-libdir=@HUNTER_PACKAGE_INSTALL_PREFIX@/lib/pkgconfig"

I don't think we need it, `PKG_CONFIG_LIBDIR` should do the trick:
- https://github.com/ruslo/hunter/blob/v0.23.198/cmake/modules/hunter_autotools_project.cmake#L122-L124

Reply:
The configure script doesn't seem to pay attention to the env var on OS X at least, it ends up using `/usr/local/lib/pkgconfig`.  It was confusing initially since the hunter autotools machinery couldn't find the `.pc` files.

Reply:
> I don't think we need it, PKG_CONFIG_LIBDIR should do the trick:

Okay, it's not about dependencies (package doesn't have dependencies).

> It was confusing initially since the hunter autotools machinery couldn't find the .pc files

The error I see:
```
[hunter ** FATAL ERROR **] File not exists:
[hunter ** FATAL ERROR **]   /Users/travis/build/ingenue/hunter/_testing/Hunter/_Base/01ac862/2080cf9/4296a0f/Build/ncursesw/Install/lib/pkgconfig/ncursesw.pc
```
- https://travis-ci.org/ingenue/hunter/jobs/554862031

Probably for macOS the destination is not `/.../lib/pkgconfig`? Where files will be installed if you will not specify `--with-pkg-config-libdir`?

Reply:
Without the option, the configure output is:

```
checking if you want to use pkg-config... yes
checking for pkg-config... /usr/local/bin/pkg-config
checking for /usr/local/bin/pkg-config library directory... checking done... /usr/local/lib/pkgconfig
checking if we should install .pc files for /usr/local/bin/pkg-config... yes
checking for suffix to add to pc-files... none
```

The final directories output by the configure are:

```
        bin directory: /Users/tstack/github/hunter/_testing/Hunter/_Base/6ecd3e4/374ced1/3a5f6cb/Build/ncursesw/Install/bin
        lib directory: /Users/tstack/github/hunter/_testing/Hunter/_Base/6ecd3e4/374ced1/3a5f6cb/Build/ncursesw/Install/lib
    include directory: /Users/tstack/github/hunter/_testing/Hunter/_Base/6ecd3e4/374ced1/3a5f6cb/Build/ncursesw/Install/include/ncursesw
        man directory: /Users/tstack/github/hunter/_testing/Hunter/_Base/6ecd3e4/374ced1/3a5f6cb/Build/ncursesw/Install/share/man
   terminfo directory: /Users/tstack/github/hunter/_testing/Hunter/_Base/6ecd3e4/374ced1/3a5f6cb/Build/ncursesw/Install/share/terminfo
 pkg-config directory: /usr/local/lib/pkgconfig
```

With the option:

```
checking if you want to use pkg-config... yes
checking for pkg-config... /usr/local/bin/pkg-config
checking for /usr/local/bin/pkg-config library directory... /Users/tstack/github/hunter/_testing/Hunter/_Base/16bc643/374ced1/3a5f6cb/Build/ncursesw/Install/lib/pkgconfig
checking if we should install .pc files for /usr/local/bin/pkg-config... yes
checking for suffix to add to pc-files... none
```

And, the summary is:

```
        bin directory: /Users/tstack/github/hunter/_testing/Hunter/_Base/16bc643/374ced1/3a5f6cb/Build/ncursesw/Install/bin
        lib directory: /Users/tstack/github/hunter/_testing/Hunter/_Base/16bc643/374ced1/3a5f6cb/Build/ncursesw/Install/lib
    include directory: /Users/tstack/github/hunter/_testing/Hunter/_Base/16bc643/374ced1/3a5f6cb/Build/ncursesw/Install/include/ncursesw
        man directory: /Users/tstack/github/hunter/_testing/Hunter/_Base/16bc643/374ced1/3a5f6cb/Build/ncursesw/Install/share/man
   terminfo directory: /Users/tstack/github/hunter/_testing/Hunter/_Base/16bc643/374ced1/3a5f6cb/Build/ncursesw/Install/share/terminfo
 pkg-config directory: /Users/tstack/github/hunter/_testing/Hunter/_Base/16bc643/374ced1/3a5f6cb/Build/ncursesw/Install/lib/pkgconfig
```

It just seems to prefer the pkg-config on the path over the env var.

Reply:
> It just seems to prefer the pkg-config on the path over the env var

I think it's an upstream bug, the `--prefix` is ignored. As I noted above it's not about `PKG_CONFIG_LIBDIR` because the package is not looking for dependencies.

The thing about `--with-pkg-config-libdir=@HUNTER_PACKAGE_INSTALL_PREFIX@/lib/pkgconfig` is that it's not designed to be used this way and works accidentally because of missing CMake feature: we have to force `configure_file` to unify the line endings:
- https://github.com/ruslo/hunter/blob/v0.23.198/cmake/modules/hunter_create_args_file.cmake#L129-L136

I've added `--with-pkg-config-libdir=` to autotools scheme. It doesn't cause an error for other packages, but only warning, so it should be fine:
- https://github.com/ruslo/hunter/commit/f0672dac60506f9d1d2e0dca1ad68b0d24ca0989